snuba/datasets/configuration/json_schema.py

```diff
@@ -515,6 +515,10 @@ def registered_class_schema(
             "type": "boolean",
             "description": "Flag to determine if migration group is optional",
         },
+        "execute_before": {
+            "type": "string",
+            "description": "The migration group this should be run directly before.",
+        },
```

Comment:
I am not following the reasoning for this. 
Migrations group are designed to be independent from each other so that you do not need to run all of them in your environment. 
Which is what we do as we have disabled groups. 
This implies that there cannot be a dependency relationship between groups.

Reply:
Well there is a dependency between groups, since the discover group can't be run before the transactions/errors group. Right now that dependency is encoded implicitly in the order of `REGISTERED_GROUPS` but that is no longer reliable with the way configurations are loaded.

Reply:
> Well there is a dependency between groups, since the discover group can't be run before the transactions/errors group. Right now that dependency is encoded implicitly in the order of REGISTERED_GROUPS but that is no longer reliable with the way configurations are loaded.

That's because the data model for discover and transaction migrations is broken. It is not true that discover depends on transactions. If we drop a column the dependency is inverted. In that case discover *must* be executed first.
So we cannot introduce that relationship in the framework knowing that it generates an impossible scenario that could not work.